File: backend-dotnet/Controllers/TasksController.cs
using backend_dotnet.Data.Dtos;
using backend_dotnet.Services;
using Microsoft.AspNetCore.Mvc;

namespace backend_dotnet.Controllers
{
    [Route("apiTask")]
    [ApiController]
    public class TasksController : ControllerBase
    {
        private TasksServices taskServices;

        public TasksController(TasksServices taskServices)
        {
            this.taskServices = taskServices;
        }

        [HttpGet("taskList")]
        public IActionResult GetTaskList([FromBody] TaskListDto filter)
        {
            try
            {
                List<TaskListDto> dtoList = taskServices.GetTasksList(filter);
                if (!dtoList.Any()) return Ok("Don't exists tasks");
                return Ok(dtoList);
            }
            catch (Exception e)
            {
                return BadRequest(e.Message);
            }
        }

        [HttpGet("{taskId}")]
        public IActionResult GetTask(string taskId)
        {
            try
            {
                TaskListDto? dto = taskServices.GetTask(taskId);
                if (dto == null) return Ok("Task " + taskId + " is not exists");
                return Ok(dto);
            }
            catch (Exception e)
            {
                return BadRequest(e.Message);
            }
        }

        [HttpPost("createtask")]
        public IActionResult CreateTask([FromBody] TasksDto dto)
        {
            try
            {
                bool isCreated = taskServices.CreateTask(dto);
                if (isCreated) return Ok("Create Task Successfully");
                return BadRequest("Create Task not Successfully");
            }
            catch (Exception e)
            {
                return BadRequest(e.Message);
            }
        }

        [HttpPut("{taskId}")]
        public IActionResult UpdateTask(string taskId, [FromBody] TasksDto dto)
        {
            try
            {
                bool isUpdated = taskServices.UpdateTask(taskId, dto);
                if (isUpdated) return Ok("Update Task Successfully");
                return BadRequest("Update Task not Successfully");
            }
            catch (Exception e)
            {
                return BadRequest(e.Message);
            }
        }

        [HttpPut("updateState/{taskId}")]
        public IActionResult UpdateState(string taskId, [FromBody] string state)
        {
            try
            {
                bool isUpdated = taskServices.UpdateState(taskId, state);
                if (isUpdated) return Ok("Update State Successfully");
                return BadRequest("Update State not Successfully");
            }
            catch (Exception e)
            {
                return BadRequest(e.Message);
            }
        }

        [HttpDelete("{taskId}")]
        public IActionResult DeleteTask(string taskId)
        {
            try
            {
                bool isDeleted = taskServices.DeleteTask(taskId);
                if (isDeleted) return Ok("Delete Task Successfully");
                return BadRequest("Delete Task not Successfully");
            }
            catch (Exception e)
            {
                return BadRequest(e.Message);
            }
        }
    }
}

File: backend-python/controllers/tasks_controller.py
from fastapi import APIRouter, Depends
from sqlalchemy.orm import Session
from database import get_db
from dtos.tasks_dto import TaskDto
from services import tasks_service

controller = APIRouter(prefix="/apiTask", tags=["Tasks"])

@controller.post("/taskList")
def get_task_list(dto: TaskDto, db: Session = Depends(get_db)):
    try:
        result = tasks_service.get_task_list(dto, db)
        if not result:
            return "Don't exists tasks"
        return result
    except Exception as ex:
        return str(ex)

@controller.get("/{task_id}")
def get_task(task_id: str, db: Session = Depends(get_db)):
    try:
        result = tasks_service.get_task(task_id, db)
        if not result:
            return "Task doesn't exists"
        return result
    except Exception as ex:
        return str(ex)

@controller.post("/createTask")
def create_task(dto: TaskDto, db: Session = Depends(get_db)):
    try:
        result = tasks_service.create_task(dto, db)
        if result is True:
            return "Task created successfully"
        if result is False:
            return "Missing required fields"
        return result
    except Exception as ex:
        return str(ex)

@controller.put("/{task_id}")
def update_task(task_id: str, dto: TaskDto, db: Session = Depends(get_db)):
    try:
        result = tasks_service.update_task(task_id, dto, db)
        if result is True:
            return "Task updated successfully"
        if result is False:
            return "Task doesn't exists"
        return result
    except Exception as ex:
        return str(ex)

@controller.put("/updateState/{task_id}")
def update_state(task_id: str, state: str, db: Session = Depends(get_db)):
    try:
        result = tasks_service.update_state(task_id, state, db)
        if result is True:
            return "State updated successfully"
        if result is False:
            return "Task doesn't exists or missing data"
        return result
    except Exception as ex:
        return str(ex)

@controller.delete("/{task_id}")
def delete_task(task_id: str, db: Session = Depends(get_db)):
    try:
        result = tasks_service.delete_task(task_id, db)
        if result is True:
            return "Task deleted successfully"
        return "Task doesn't exists"
    except Exception as ex:
        return str(ex)
